charts/ark-solr/templates/statefulset.yaml

```diff
@@ -0,0 +1,63 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "solr.fullname" . }}-statefulset
+  labels:
+    {{- include "solr-helm-charts.labels" . | nindent 6 }}
+  namespace: {{ .Release.Namespace | quote }}
+spec:
+  replicas: 1
+  selector: 
+    matchLabels:  
+      {{- include "solr-helm-charts.labels" . | nindent 6 }}
+  serviceName: {{ include "solr.fullname" . }}-headless
+  template:
+    metadata:
+      name: ark-solr
+      labels:
+        {{- include "solr-helm-charts.labels" . | nindent 8 }}
+    spec:
+      initContainers:
+      containers:
+      - name: solr
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        ports:
+        - containerPort: 8983
+          name: solr-port
+          protocol: TCP
+        command:
+        livenessProbe:
+        readinessProbe: 
+        resources:
+        volumeMounts:
+        - name: solr-data
+          mountPath: /data
+        - name: solr-conf
+          mountPath: /opt/solr/server/solr/configsets/_default/conf
+      volumes:
+        - name: solr-conf
+          configMap: 
+            name: {{ include "solr.configmap-name" . }}
+        {{- if and .Values.persistence.enabled .Values.persistence.existingClaim }}
+        - name: solr-data
+          persistentVolumeClaim:
+            claimName: {{ printf "%s" (tpl .Values.persistence.existingClaim .) }}
+        {{- else if not .Values.persistence.enabled }}
+        - name: solr-data
+          emptyDir: {}
+        {{- end -}}
+  {{- if and .Values.persistence.enabled (not .Values.persistence.existingClaim) }}
+  volumeClaimTemplates:
+    - metadata:
+        name: solr-data
+      spec:
+        accessModes:
+          {{- range .Values.persistence.accessModes }}
+            - {{ . | quote }}
+          {{- end }}
+        resources:
+          requests:
+            storage: {{ .Values.persistence.size | quote }}
+        storageClassName: {{ .Values.persistence.storageClassName | default "gp2"}}
```

Comment:
Please set the default to `standard`. It's better to be generic rather than specific for default values.

Reply:
@fabricetriboix  I don't know what `standard` storageclass. When  EBS drivers installed it creates a default gp2 storageclass that is why I kept gp2 as a default value.

Reply:
@naveen-armedia `gp2` is specific to AWS, while `standard` is generic

Reply:
@fabricetriboix okay removed storageClassName. So it will create pvc with its default storageclass based on the platform it is running.

Reply:
I don't think that's a good idea. We should be able to specify the storage class as a variable. Whatever is calling the helm program must be able to influence that parameter. Or are you saying that in such a case whatever is calling the helm chart would be in charge of creating a PVC first?